libs/portals/admin/ids-admin/src/components/Application/ApplicationsUrl.tsx

```diff
@@ -0,0 +1,87 @@
+import { Input, Stack, Text } from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import React, { useEffect, useState } from 'react'
+import { m } from '../../lib/messages'
+import ContentCard from './ContentCard'
+import { AuthApplicationApplicationUrlList } from './Application.loader'
+
+interface ApplicationsUrlProps {
+  applicationUrls: AuthApplicationApplicationUrlList
+}
+const ApplicationsUrl = ({ applicationUrls }: ApplicationsUrlProps) => {
+  const { formatMessage } = useLocale()
+  const [appUrls, setAppUrls] = useState(applicationUrls)
+
+  // Generic onChange handler, name in input will need to match object name to change
+  const onChangeURLS = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    setAppUrls((prev) => ({
+      ...prev,
+      [event.target.name]: event.target.value.split(', '),
+    }))
+  }
+
+  useEffect(() => {
+    setAppUrls(applicationUrls)
+  }, [applicationUrls])
+
+  return (
+    <ContentCard
+      title={formatMessage(m.applicationsURLS)}
+      onSave={(saveOnAllEnvironments) => {
+        console.log('saveOnAllEnvironments: ', saveOnAllEnvironments, appUrls)
+      }}
+    >
+      <Stack space={3}>
+        <Stack space={1}>
+          <Input
+            name="callbackUrls"
+            type="text"
+            size="sm"
+            label={formatMessage(m.callbackUrl)}
+            textarea
+            rows={4}
+            onChange={onChangeURLS}
+            backgroundColor="blue"
+            value={appUrls.callbackUrls.join(', ')}
```

Comment:
Should be able to provide the right `defaultValue` and just handle the rest in the action/zod.